### PR TITLE
Events - Add "turretOpticsMode" Player EH

### PR DESCRIPTION
**Pull request by hypoxia125**
**When merged this pull request will:**
- Add player event handler to detect when the player changes turret zoom

Requires Arma 3 2.10

### CODE REVIEW

**Comment by hypoxia125** [comment]
Oops. Didn't realize that BI already added an engine version... Closing

File: addons/events/fnc_addPlayerEventHandler.sqf [hunk 1]
```diff
@@ -103,6 +104,12 @@ private _id = switch (_type) do {
         };
         [QGVAR(turretEvent), _function] call CBA_fnc_addEventHandler // return id
     };
+    case "turretopticsmode": {
+        if (_applyRetroactively) then {
+            [GVAR(oldUnit), getTurretOpticsMode GVAR(oldUnit)] call _function;
```
Comment (PabstMirror) [inline addons/events/fnc_addPlayerEventHandler.sqf:109]:
```suggestion
            [GVAR(oldUnit), getTurretOpticsMode GVAR(oldUnit), -1] call _function;
```

File: addons/events/fnc_playerEvent.sqf [hunk 2]
```diff
@@ -44,7 +44,7 @@ private _state = [
     _unit, group _unit, leader _unit,
     currentWeapon _unit, currentMuzzle _unit, currentWeaponMode _unit,
     getUnitLoadout _unit, _vehicle, _turret, _vehicle currentWeaponTurret _turret,
-    currentVisionMode _controlledEntity, cameraView
+    currentVisionMode _controlledEntity, cameraView, getTurretOpticsMode _unit
```
Comment (PabstMirror) [inline addons/events/fnc_playerEvent.sqf:47]: I wonder if this should be `_unit` or `focusOn` for UAVs
will test

Comment (PabstMirror) [reply addons/events/fnc_playerEvent.sqf:47]: doesn't seem to matter
`getTurretOpticsMode focuson` = `getTurretOpticsMode player`

**Pull request merged**